setup/git/000_EnvSetup.rb

```diff
@@ -85,44 +66,63 @@
 action=nocheck
 # Install to the default base directory.
 basedir=default
-END
-    on host, 'pkgadd -a /root/shutupsolaris -d http://get.opencsw.org/now all'
-    on host, '/opt/csw/bin/pkgutil -U all'
-    on host, '/opt/csw/bin/pkgutil -i -y ruby19_dev'
-    on host, '/opt/csw/bin/pkgutil -i -y ruby19'
-    on host, 'ln -sf /opt/csw/bin/gem19 /usr/bin/gem'
-    on host, 'ln -sf /opt/csw/bin/ruby19 /usr/bin/ruby'
+      END
+      on host, 'pkgadd -d http://get.opencsw.org/now -a /root/shutupsolaris -n all'
+      on host, '/opt/csw/bin/pkgutil -U all'
+      on host, '/opt/csw/bin/pkgutil -y -i pkgutil'
+      on host, '/opt/csw/bin/pkgutil -y -i coreutils'
+      on host, '/opt/csw/bin/pkgutil -y -i git'
+      on host, '/opt/csw/bin/pkgutil -y -i gtar'
+    end
+    host.add_env_var('PATH', '/opt/puppetlabs/puppet/bin:/opt/csw/bin')
+  elsif host['platform'] =~ /osx/
+    host.add_env_var('PATH', '/opt/puppetlabs/puppet/bin:/usr/local/bin')
+    #host.add_env_var('PATH', '/opt/puppetlabs/puppet/bin')
+  elsif host['platform'] =~ /windows/
+    # might also need this one:
+    # /cygdrive/c/ProgramFiles64Folder/PuppetLabs/Puppet/puppet/bin
+    host.add_env_var('PATH', '/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Puppet/sys/ruby/bin:/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Puppet/bin')
+  else
+    host.add_env_var('PATH', '/opt/puppetlabs/puppet/bin')
   end
 end
 
+dev_builds_url = ENV['DEV_BUILDS_URL'] || 'http://builds.delivery.puppetlabs.net'
+branch = 'master'
+
+runtime_dir = Dir.mktmpdir('puppet-runtime')
+`git clone --depth 1 git@github.com:puppetlabs/puppet-runtime.git #{runtime_dir}`
+
+runtime_tag = nil
+Dir.chdir '/tmp/puppet-runtime' do
+  runtime_tag = `git describe --abbrev=0`.chomp
+end
+
+runtime_url = "#{dev_builds_url}/puppet-runtime/#{runtime_tag}/artifacts/"
+runtime_prefix = "agent-runtime-#{branch}-#{runtime_tag}."
+runtime_suffix = ".tar.gz"
```

Comment:
I want to use artifactory to give us this information rather than constructing it here, but I haven't figured out the best way to do that yet

```
$ jfrog rt s generic__local/*agent-runtime*master*201806220*windows*.tar.gz
[Info] Searching artifacts...
[Info] Found 2 artifacts.
[
  {
    "path": "generic__local/development/puppet-runtime/201806220/windows-x64/agent-runtime-master-201806220.windows-2012r2-x64.tar.gz",
    "props": {
      "": ""
    }
  },
  {
    "path": "generic__local/development/puppet-runtime/201806220/windows-x86/agent-runtime-master-201806220.windows-2012r2-x86.tar.gz",
    "props": {
      "": ""
    }
  }
]
```